### PR TITLE
Test script for CCDB key rotation

### PR DESCRIPTION
**Pull request by andreas-kupries**
## Description

- Added a script performing CCDB key rotation (updating the key label in the deployment, waiting for reconfiguration, triggering the rotation errand, waiting for the errand to complete)
- :warning: :construction: Integration with the concourse post-publish pipeline. :construction: :warning: 

## Motivation and Context

  - https://github.com/SUSE/kubecf/issues/299
  - https://jira.suse.com/browse/CAP-1164

## How Has This Been Tested?

  - Currently manually in a minikube SA setup.
  - After integration with post-publish running the PR in some concourse.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by jimmykarily** [comment]
This PR will have conflicts with this one: https://github.com/SUSE/kubecf/pull/401 . The pipeline was heavily refactored (and renamed) in order to also test PRs and replace the (no longer working) GitLab pipeline. Looking at this PR, it shouldn't be too difficult to adjust but let ping me or @mudler if you face issues.

File: .concourse/rotate-ccdb-keys-test.sh [hunk 1]
```diff
@@ -0,0 +1,235 @@
+#!/bin/bash
```
Comment (mudler) [inline .concourse/rotate-ccdb-keys-test.sh]: Maybe it's better for this file to not live under `.concourse/` so it is possible to run the test also outside of the ci context

**Comment by mudler** [comment]
Wouldn't make sense to have a `bazel` target to run extra group of tests? In this way we can have a step in the pipeline that can cover more cases as needed, and at the same time we can run them easily locally too.

**Comment by andreas-kupries** [comment]
@jimmykarily Will have :eyes: on #401 and check how that interacts with my stuff.
@mudler How about moving it to under `testing/` ? Maybe a `testing/scripts/` directory.
Bazel target likely a good idea also. Can however be added later. IOW when the step is working in the pipeline.

**Comment by f0rmiga** [comment]
@andreas-kupries `//testing/ccdb_key_rotation` sounds like a good target for running the testing script.

**Comment by andreas-kupries** [comment]
@jimmykarily Ok. Only one commit on this branch, the last (dad5129) touches the pipeline spec file. Undoing that and I can rebase my work on top of #401, shuffle things as suggested, and then redo an integration of the new step.

File: .concourse/rotate-ccdb-keys-test.sh (hunk 1 above)
Comment (andreas-kupries) [reply .concourse/rotate-ccdb-keys-test.sh]: :+1:

**Comment by andreas-kupries** [comment]
Rebased this PR to be on top of #401. Shuffled script into a different place, and redid the pipeline integration. The commit 5a7220a07ed00313c6ee1016d1d2fecf786bf660 fixes some issues in the base pipeline (borken anchoring/references in the first steps of the PR pipeline).

**Comment by jimmykarily** [comment]
We pulled all the commits from this PR to this one: https://github.com/SUSE/kubecf/pull/401 . We will address the comments on #401 and merge to master. Improvements and additional fixes can happen on a new branch.

**Pull request closed without merging**